spec/lib/brexit_checker/criteria/filter_spec.rb

```diff
@@ -0,0 +1,74 @@
+require 'spec_helper'
+
+RSpec.describe BrexitChecker::Criteria::Filter do
+  describe "#call" do
+    let(:question1) do
+      FactoryBot.build :brexit_checker_question, options: [
+        FactoryBot.build(:brexit_checker_option, value: 'c1', sub_options: [
+          FactoryBot.build(:brexit_checker_option, value: 'c2')
+        ])
+      ]
+    end
+
+    let(:question2) do
+      FactoryBot.build :brexit_checker_question, options: [
+        FactoryBot.build(:brexit_checker_option, value: 'c3')
+      ]
+    end
+
+    let(:question3) do
+      FactoryBot.build :brexit_checker_question, options: [
+        FactoryBot.build(:brexit_checker_option, label: "Option group", sub_options: [
+          FactoryBot.build(:brexit_checker_option, value: 'c4')
+        ])
+      ]
+    end
+
+    before do
+      allow(question1).to receive(:show?) { true }
+      allow(question2).to receive(:show?) { false }
+      allow(question2).to receive(:show?).with(array_including('c2')) { true }
+      allow(question3).to receive(:show?) { false }
+      allow(question3).to receive(:show?).with(array_including('c3')) { true }
+
+      allow(BrexitChecker::Question).to receive(:load_all) do
+        [question1, question2, question3]
+      end
+    end
+
+    it "preserves criteria for questions without dependencies" do
+      criteria = subject.call(%w(c1))
+      expect(criteria).to include('c1')
+    end
+
+    it "filters criteria that are not present in any question" do
+      criteria = subject.call(%w(c5))
+      expect(criteria).to_not include('c5')
+    end
+
+    it "filters criteria for questions with unmet dependencies" do
+      criteria = subject.call(%w(c3))
+      expect(criteria).to_not include('c3')
+    end
+
+    it "preserves criteria for questions with met dependencies" do
+      criteria = subject.call(%w(c1 c2 c3))
+      expect(criteria).to include('c3')
```

Comment:
For these example, could we have expectations that match the full return value? 
```suggestion
      expect(criteria).to contain_exactly('c2', 'c3')
```
I feel we aren't catching potential incorrect return values i.e. an array with only 'c3'

Reply:
I think understand the reason behind the question: it's unclear what other elements might be in the array. For me, the most important thing is that the expectation is clearly tied to the 'it' statement. So, in this case: "preserves criteria for questions with met dependencies" is referring specifically to 'c3', and 'c2' is irrelevant (but covered by other specs).

I've thought of a compromise that could satisfy both needs, but it would make the specs a bit longer to implement. Before I do that, can I check if you feel this is a real issue?

Reply:
We discussed in person the need for checking all elements in the return value as the tests don't cover scenarios where sets of questions include those without out and met dependencies.

Reply:
Done. I thought about it a bit more and simplified some of the other matchers too 👍.